manual/statements/built/README.md

```diff
@@ -1,5 +1,268 @@
+                .ttl("t").as("id_ttl").writeTime("t")
+                .from("test", "test");
+```
+
+You can also cast the value of the given column to another type by using the `cast`
+function,
+[specifying](https://docs.datastax.com/en/drivers/java/3.5/com/datastax/driver/core/querybuilder/QueryBuilder.html#cast-java.lang.Object-com.datastax.driver.core.DataType-)
+the column for which it should be performed & to what type it should be casted.
+
+#### Specifying conditions
+
+Selection of data rarely happen on the whole table - in most cases, people are interested
+for particular rows, located in one or several partitions.  Conditions are specified by
+using the `where` call, like this:
+
+```java
+BuiltStatement selectOne = QueryBuilder.select().from("test", "test")
+        .where(QueryBuilder.eq("id", 1));
+```
+
+The `where` function accept the `Clause` object that is generated by calling
+`QueryBuilder`'s
+[functions](https://docs.datastax.com/en/drivers/java/3.5/com/datastax/driver/core/querybuilder/QueryBuilder.html#eq-java.lang.Iterable-java.lang.Iterable-),
+such as, `eq`, `ne`, `lt`, `in`, `contains`, `notNull`, etc.  In most cases, these
+functions receive 2 arguments - the name of the column, and value to compare, but there
+are also variants that receive 2 iterables for columns & values separately.
+
+Note: as query are becoming more complex, repeating `QueryBuilder` at all places will make
+code less readable.  In this case you can import all (or only required) static functions
+for code simplification (this is rewritten example from above):
+
+```java
+import static com.datastax.driver.core.querybuilder.QueryBuilder.*;
+
+//...
+
+BuiltStatement selectOne = QueryBuilder.select().from("test", "test")
+        .where(eq("id", 1));
+```
+
+In case if you need to specify complex condition you can put additional clauses into chain
+of the `and` calls that accepts the same clauses as `where`:
+
+```java
+BuiltStatement select = QueryBuilder.select().from("test", "test")
+        .where(eq("id", "1")).and(eq("txt", "test"));
+```
+
+#### Other selection options
+
+For `SELECT` statements you can also specify a lot of different options: 
+ - `allowFiltering` generates corresponding `ALLOW FILTERING` part of query (***only use if you know what you're doing!***);
+ - `limit` & `perPartitionLimit` allows to specify amount of data to fetch;
+ - `groupBy` performs group of data;
+ - `orderBy` allows to specify sorting direction for specified clustering columns;
+ 
+This very "artificial" example shows the use for some of them:
+
+```java
+BuiltStatement selectOne = QueryBuilder.select().from("test")
+        .where(QueryBuilder.eq("id", 1)).limit(1).allowFiltering()
+        .perPartitionLimit(1).orderBy(desc("id"));
+```
+
+### Inserting data
+
+Insertion of data is straightforward - you're specifying the target table in call to
+`insertInto`, and then provide values to insert either by chaining the several calls to
```

Comment:
Suggestion: " either by chaining several calls to the `value` function,  or by using the `values` function and passing lists or arrays of column names and their corresponding values".